File: leetcode/python/0012_integer_to_roman.py
class Solution:
  def intToRoman(self, num: int) -> str:
    digits = [
      (1000, "M"),
      (900, "CM"),
      (500, "D"),
      (400, "CD"),
      (100, "C"),
      (90, "XC"),
      (50, "L"),
      (40, "XL"),
      (10, "X"),
      (9, "IX"),
      (5, "V"),
      (4, "IV"),
      (1, "I")
    ]
    res = ""

    # loop through digits using i:
    # if num > digits[i][0]: res += digits[i][1] * (num // digits[i][0]), num -= digits[i][0] * (num // digits[i][0])

    for i in range(len(digits)):
      if num >= digits[i][0]:
        res += digits[i][1] * (num // digits[i][0])
        num -= digits[i][0] * (num // digits[i][0])
      if num == 0: break

    return res     

File: leetcode/cpp/0012_integer_to_roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        string ret = "";
        vector<string> romans = { "I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"};
        vector<int> values = { 1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000 };
        int index = values.size() - 1;

        while (num > 0) {
            while (values[index] <= num) {
                ret += romans[index];
                num -= values[index];
            }

            index--;
        }

        return ret;
    }
};